Version Control Systems

kiranvasudev opened the pull request:
I have added the schedule to the readme of the repo so that the contributors can have a look at it. Also, please tell me if there is something I am missing in the VCS slideset. I will add content for ssh soon.

argenos commented:
Unfortunately we can't review binary (odt files) here, if it's not too much trouble, could we switch to LaTeX?

argenos commented on README.md:
```diff
@@ -1,5 +1,8 @@
 # Foundations Course
 
+## Tentative Schedule
+[FC Schedule](https://docs.google.com/spreadsheets/d/1LsfJWwRvhkPgQox0MIep3S6L5Wi7NF0fC4IZSB0zUDQ/edit?usp=sharing)
```
Is this link view-only?

kiranvasudev replied:
yes it is

argenos commented on README.md:
```diff
@@ -1,5 +1,8 @@
 
+## Tentative Schedule
+[FC Schedule](https://docs.google.com/spreadsheets/d/1LsfJWwRvhkPgQox0MIep3S6L5Wi7NF0fC4IZSB0zUDQ/edit?usp=sharing)
+
```
How is the timing for the presentation, do you think it will take 30 minutes?
Seeing the contents, I think it's doable to switch to LaTeX, could you please do that and add the changes?

kiranvasudev replied:
To be honest, from whatever I have read about VCS, that is most of the introduction given. Unless I need to go into Subversion or other VCS. Please do let me know if you find something missing, I will add it. The current presentation should be around 15-20 mins.

argenos replied:
I would add a slide about what the differences between different VCSs and why people are using git (which is a nice Segway to the next session)

kiranvasudev replied:
The main differences are at an abstract level(centralized vs distributed). I could add some more points to the centralized and distributed VCS slides though

kiranvasudev replied:
From what I have read, some of the features were better in git, for example branching. As of today, there is not much difference between the most popular VCSs and git. Please correct me if I am wrong.

argenos replied:
Sounds good! I think those two points should definitely be covered in the slides

kiranvasudev commented:
Hello. I am sorry about the delay. I got caught up with some important work. I have updated the slideset. I will be explaining the differences between the main DVCS and CVCSs. I have added the LaTeX file.

argenos commented on content/version_control/VCS.tex:
```diff
@@ -0,0 +1,322 @@
+  % You might wish to add the option [pausesections]
+\end{frame}
+
+% Section and subsections will appear in the presentation overview
```
Please remove the commented out stuff.

argenos commented on content/version_control/VCS.tex:
```diff
@@ -0,0 +1,322 @@
+\end{frame}	
+
+\begin{frame}{Types of VCS}{Git vs SVN}
+\subsection{Git vs SVN}
```
The message here is a little confusing. Group this into pros and cons of git and SVN so they are a little bit more clear. 
In the end, the message the students should get is "Why do I need to use git?"

kiranvasudev commented:
Hello @argenos 
  I have made the requested changes. Could you please review and close this pull request if everything is good? I have added the schedule for the FC in one of the earlier commits which cannot be seen by anyone unless this is closed or you adding it. I will open another pull request for the git slides.

The changes files are only VCSs. I will work on the Git slides again(on LaTeX) and upload them tomorrow night. Please excuse my delay. Some important work has come up. Thank you.

argenos commented:
I've cherry-picked your README commit. In the future, please try to keep changes grouped in different branches (and different pull requests).

argenos commented on content/version_control/VCS.tex:
```diff
@@ -0,0 +1,158 @@
+\end{frame}
+
+\begin{frame}{Take home message}
+\subsection{Take home message}
```
After reading this slide, it seems that it would be easier and better to use SVN, and all in all, I'm not sure it is clear to students why they should use VCS at all. 
![File naming](https://doctorfreelance.com/wp-content/uploads/2016/05/file-naming-768x566.png)

kiranvasudev commented:
@argenos, thank you for the Git exercise ideas. What I am doing is combining the current exercise sheet and your ideas as well. I will push the pdf latest today.

alex-mitrevski commented on content/version_control/VCS.tex:
```diff
@@ -0,0 +1,174 @@
+		    In distributed VCS, everyone has a local copy of the entire work’s history
+		  }
+		  \item {
+		    Each repository is as good as the other
```
What does this mean?

alex-mitrevski commented on content/version_control/VCS.tex:
```diff
@@ -0,0 +1,174 @@
+		    Each repository is as good as the other
+		  }
+		  \item{
+			Prone to central server crash
```
It's actually robust to server crashes; that's the benefit of decentralised version control...

alex-mitrevski commented on content/version_control/VCS.tex:
```diff
@@ -0,0 +1,174 @@
+			Your local copy is a repository, and you can commit to it and get all benefits of source control - Offline Source Control.
+		  }	  
+		  \item{
+			Mozilla Firefox uses this kind of versioning.  
```
I think the message will be clearer if you say "Millions of projects on GitHub use this kind of versioning" rather than just giving a single example.

argenos commented on content/version_control/VCS.tex:
```diff
@@ -0,0 +1,174 @@
+\begin{frame}{Take home message}
+\section{Take home message}
+\begin{itemize}
+    \item There is not much difference between the most popular VCSs. It ultimately comes down to the use case.
```
Sorry to be a pain in the ass! I think these are the last few changes.

I was on a bit of a rush last time I wrote. What I meant was, that the main point of this slide set should be about *why students/we should use VCS*. I think these bullets are fine, but maybe add a slide before this one with the reasoning behind this, even the following image with explanations will probably suffice:
![file naming](https://doctorfreelance.com/wp-content/uploads/2016/05/file-naming-768x566.png)

Once we accept the merge request, maybe I'll add a few bullet points to explain why git vs svn, hopefully, that will make it clearer

kiranvasudev replied:
I think slide 3 says this. I will add this picture to the slide though

argenos replied:
And that's fine, but the take home message should also focus on this. So I'd add it at the end

argenos commented on content/version_control/VCS.tex:
```diff
@@ -0,0 +1,174 @@
+  \begin{thebibliography}{10}
+
+  \bibitem{website}
+	\href{https://www.appfusions.com/display/StashSCMImporter/CVCS+vs.+DVCS+In+a+Nutshell
```
For some reason when I render the pdf, your links are not showing up. 
Please also commit the pdf, for easier access and review!

kiranvasudev replied:
You mean the references are not clickable?

argenos replied:
No, the links are not shown in the references slide. They should also be displayed.

argenos commented on content/version_control/VCS.tex:
```diff
@@ -0,0 +1,174 @@
+\end{itemize}
+\textbf{Pros of using Git}
+\begin{itemize}
+    \item Git is much faster than SVN as they do not have to keep communicating with a central server when changes are made.
```
I think one of the biggest advantages of git is merging, have you looked into this?

kiranvasudev replied:
I will explain that point with the 3rd pro of Git

argenos replied:
I think it's better if the idea of why is it easier to merge using git than svn should be mentioned here, and the practical aspects of merging in the git session

argenos commented on content/version_control/VCS.tex:
```diff
@@ -0,0 +1,174 @@
+\textbf{Pros of using SVN}
+\begin{itemize}
+    \item Subversion has better GUIs than Git
+    \item Working through versions in Git is tougher than SVN. Git uses SHA-1 hashes while SVN uses sequential revision numbers. 
```
Works well with binary files, which is not the case for git.

alex-mitrevski commented on content/version_control/VCS.tex:
```diff
@@ -0,0 +1,174 @@
+
+\author{Kiran Vasudev\inst{1}}
+
+\institute[Universities of Somewhere and Elsewhere] 
```
Can you please replace "Universities of Somewhere and Elsewhere" with "HBRS?"

alex-mitrevski commented on content/version_control/VCS.tex:
```diff
@@ -0,0 +1,174 @@
+\end{columns}
+\end{frame}	
+
+\begin{frame}[allowframebreaks]{Types of VCS}{Git vs SVN}
```
I think it would be a good idea to at least mention Mercurial.

kiranvasudev commented:
Requested changes have been made

argenos commented:
I still see a few changes that were not addressed, we can merge after that!

kiranvasudev commented:
Have added the slides that help explain the branching models. I will do most of the explaining and the slides contain the images.

kiranvasudev commented:
I think to clearly understand how different the branching is, those slides need to be there. It would be too difficult to summarize it in one bullet point. I will change the VCS image soon
